### DISCUSSION TITLE
BUILD FAIL: build e3sm failed

### DISCUSSION THREAD

**Discussion by PappuP** [E3SM-Project/E3SM#5681] [E3SM model help]
I am a graduate student of Atmospheric Sciences at UIUC. I am trying to port E3SM-2.1 to our local machine, and I need help. I have attached config_machine(mach=keeling), config_batch(mach=keeling), and keeling.cmake files that I have edited. 

Using these, I create, build, and run cases for compsets A and X, which are test compsets. 

**However, when I tried to build a case using compset F (F1850C5L45BGC) which is a component of the atmosphere model, I got the error message :**
​​​​​​​​​​​​​​​​​​​​​​```````````````````````````````
ERROR: CMake Error at cmake/kokkos_compiler_id.cmake:153 (MESSAGE):
  Compiler not supported by Kokkos.  Required compiler versions:

      Clang      4.0.0 or higher
      GCC        5.3.0 or higher
      Intel     17.0.0 or higher
      NVCC      9.2.88 or higher
      HIPCC      3.8.0 or higher
      PGI         17.4 or higher

**Additionally, when I tried to build a case using compset I (I1850ELM) which is a component of the land model, I got the error message:**
````````````````````​
ERROR: BUILD FAIL: build e3sm failed, cat /data/keeling/a/pappup2/a/E3SM/l1/run/l1/bld/e3sm.bldlog.230511-112858

farm/v1eca/v1ecaNlayer/v1ecaBGCReactionsType.F90(54): error #6580: Name in only-list does not exist.   [BETR_BOUNDS_TYPE]
  use BeTR_decompMod    , only : betr_bounds_type
---------------------------------^
/data/keeling/a/pappup2/E3SM2.1/E3SM/components/elm/src/external_models/sbetr/src/Applications/soil-farm/v1eca/v1ecaNlayer/v1ecaBGCReactionsType.F90(2351): warning #6843: A dummy argument with an explicit INTENT(OUT) declaration is not given an explicit value.   [BETR_STATUS]
       betrtracer_vars, biophysforc,  tracerstate_vars, betr_status)
--------------------------------------------------------^
compilation aborted for /data/keeling/a/pappup2/E3SM2.1/E3SM/components/elm/src/external_models/sbetr/src/Applications/soil-farm/v1eca/v1ecaNlayer/v1ecaBGCReactionsType.F90 (code 1)
Target CMakeFiles/lnd.dir/__/__/elm/src/external_models/sbetr/src/Applications/soil-farm/v1eca/v1ecaNlayer/v1ecaBGCReactionsType.F90.o built in 0.459578 seconds
gmake[2]: *** [cmake/lnd/CMakeFiles/lnd.dir/__/__/elm/src/external_models/sbetr/src/Applications/soil-farm/v1eca/v1ecaNlayer/v1ecaBGCReactionsType.F90.o] Error 1
gmake[2]: Leaving directory `/data/cristi/a/pappup2/E3SM/l1/run/l1/bld/cmake-bld'
gmake[1]: *** [cmake/lnd/CMakeFiles/lnd.dir/all] Error 2
gmake[1]: Leaving directory `/data/cristi/a/pappup2/E3SM/l1/run/l1/bld/cmake-bld'
gmake: *** [all] Error 2
real 491.36
user 445.82
sys 30.77
`````````````````````````````````
I anticipate help so that I can successfully build and run cases for any compset. Feel free to ask for any further details about it if you need them.
[config_batch.xml.txt](https://github.com/E3SM-Project/E3SM/files/11454727/config_batch.xml.txt)
[keeling.cmake.txt](https://github.com/E3SM-Project/E3SM/files/11454730/keeling.cmake.txt)
[config_machines.xml.txt](https://github.com/E3SM-Project/E3SM/files/11454731/config_machines.xml.txt)

[kokkos.bldlog.230510-144103.txt](https://github.com/E3SM-Project/E3SM/files/11454700/kokkos.bldlog.230510-144103.txt)
[e3sm.bldlog.230511-112858.txt](https://github.com/E3SM-Project/E3SM/files/11454703/e3sm.bldlog.230511-112858.txt)

[software_environment.txt](https://github.com/E3SM-Project/E3SM/files/11468087/software_environment.txt)

**Comment by PappuP** [comment]
Solved these using gnu compiler instead of intel

**Comment by rljacob** [comment]
Your software_environment.txt says you're using intel v 15.0.3.  That is definitely to old.   Try to upgrade to v19 or later.
